client/signup/main.jsx

```diff
@@ -185,13 +191,13 @@ const Signup = React.createClass( {
 	componentDidMount() {
 		debug( 'Signup component mounted' );
 		SignupProgressStore.on( 'change', this.loadProgressFromStore );
-		SignupProgressStore.on( 'change', this.loadDependenciesFromStore );
+		this.unsubscribeSignupDepStoreListener = SignupDependencyStore.reduxStore.subscribe( this.loadDependenciesFromStore );
```

Comment:
If there are no changes, you can do a simple reference equality check since Redux will return the same object if state was not updated.

I was thinking of the overall pattern we'd like to have with refactoring existing code. Our redux tree is growing so we will have more events firing off. Even redux persistence which listens to store updates is throttled at the moment.